applications/API/models.py

```
@@ -0,0 +1,626 @@
+        choices=STATUS.choices,
+        db_index=True,
+    )
+
+    def save(self, *args, **kwargs):
+        # Only generate reference_id if it is not set and pk is None (new object)
+        if self.pk is None and not self.reference_id:
+            # Save first to get a PK
+            super().save(*args, **kwargs)
+            # Generate a unique reference_id
+            while True:
+                candidate = f"{random.choice(INSULT_REFERENCE_ID_PREFIX_OPTIONS)}_{encode_base64(self.pk)}"
```

Ruff (0.12.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)